commit 1b929c6f3ac51372914c4c0d49be8f27caee5ca5
Author: Justus Winter <4winter@informatik.uni-hamburg.de>
Date:   Thu Jan 16 14:47:38 2014 +0100

    vm: remove the declaration of memory_object_create_proxy
    
    It is not clear to me why the declaration was put there in the first
    place.  It is not used anywhere, and it conflicts with the declaration
    generated by mig.
    
    * vm/memory_object_proxy.h (memory_object_create_proxy): Remove declaration.

diff --git a/vm/memory_object_proxy.h b/vm/memory_object_proxy.h
index 3de6996..dc0ea74 100644
--- a/vm/memory_object_proxy.h
+++ b/vm/memory_object_proxy.h
@@ -30,17 +30,6 @@
 
 extern void memory_object_proxy_init (void);
 extern boolean_t memory_object_proxy_notify (mach_msg_header_t *msg);
-extern kern_return_t memory_object_create_proxy (ipc_space_t space,
-                                                 vm_prot_t max_protection,
-                                                 ipc_port_t *object,
-                                                 natural_t object_count,
-                                                 vm_offset_t *offset,
-                                                 natural_t offset_count,
-                                                 vm_offset_t *start,
-                                                 natural_t start_count,
-                                                 vm_offset_t *len,
-                                                 natural_t len_count,
-                                                 ipc_port_t *port);
 extern kern_return_t memory_object_proxy_lookup (ipc_port_t port,
                                                  ipc_port_t *object,
                                                  vm_prot_t *max_protection);
commit 62ca925c9a6b0d12bdf17ac4ab93f5434575cedc
Author: Justus Winter <4winter@informatik.uni-hamburg.de>
Date:   Thu Jan 16 14:50:31 2014 +0100

    kern: include the mig-generated server headers in ipc_kobject.c
    
    GNU MIG recently gained support for emitting x_server_routine
    declarations in the generated server header file.  Using this
    declaration, the x_server_routine functions can be inlined into the
    ipc_kobject_server function.
    
    * kern/ipc_kobject.c: Include the mig-generated server headers.

diff --git a/kern/ipc_kobject.c b/kern/ipc_kobject.c
index 37d4eb9..13af820 100644
--- a/kern/ipc_kobject.c
+++ b/kern/ipc_kobject.c
@@ -49,6 +49,19 @@
 #include <vm/memory_object_proxy.h>
 #include <device/ds_routines.h>
 
+#include <kern/mach.server.h>
+#include <ipc/mach_port.server.h>
+#include <kern/mach_host.server.h>
+#include <device/device.server.h>
+#include <device/device_pager.server.h>
+#include <kern/mach4.server.h>
+#include <kern/gnumach.server.h>
+#include <kern/experimental.server.h>
+
+#if MACH_DEBUG
+#include <kern/mach_debug.server.h>
+#endif
+
 #if	MACH_MACHINE_ROUTINES
 #include <machine/machine_routines.h>
 #endif
@@ -146,18 +158,6 @@ ipc_kobject_server(request)
 	 * to perform the kernel function
 	 */
     {
-	extern mig_routine_t	mach_server_routine(),
-				mach_port_server_routine(),
-				mach_host_server_routine(),
-				device_server_routine(),
-				device_pager_server_routine(),
-				mach4_server_routine(),
-				gnumach_server_routine(),
-				experimental_server_routine();
-#if	MACH_DEBUG
-	extern mig_routine_t	mach_debug_server_routine();
-#endif
-
 #if	MACH_MACHINE_ROUTINES
 	extern mig_routine_t	MACHINE_SERVER_ROUTINE();
 #endif
